src/app/marathon/schedule-management/edit/edit.component.ts

```diff
@@ -77,10 +231,292 @@ export class EditComponent implements OnInit {
           message: `Something went wrong: ${err.message}`,
           duration: 5000,
           position: 'is-right',
-          color: 'is-warning'
+          color: 'is-warning',
         };
         this.toastr.open(alertConfig);
       },
     });
   }
+
+  async loadAllAvailabilities() {
+    const allAvailabilities = await firstValueFrom(
+      this.submissionService.availabilities(this.marathonId)
+    );
+
+    this.allAvailabilities = allAvailabilities;
+
+    const entries = Object.entries<Availability[]>(allAvailabilities)
+      .sort((a, b) => a[0].localeCompare(b[0]));
+
+    for (const [username, availabilities] of entries) {
+      const contentName = availabilities.length ? availabilities[0].username : username;
+
+      this.availabilitiesGroups.add({
+        id: username,
+        content: contentName,
+      });
+
+      availabilities.forEach((availability, index) => {
+        this.availabilitiesItems.add({
+          id: username + index,
+          group: username,
+          start: availability.from,
+          end: availability.to,
+          content: '',
+        });
+      });
+    }
+
+    this.availabilitiesGroups.flush();
+    this.availabilitiesItems.flush();
+  }
+
+  async loadAvailabilitiesForRunner(userId: number) {
+    // TODO: do not load if we have a user in the array already.
```

Comment:
Can be easily done by storing a map of user id -> username, but there's probably an easier way